inc/infoutils.php

```diff
@@ -22,20 +28,16 @@ function checkUpdateMessages(){
 
     $cf = getCacheName($updateVersion, '.updmsg');
     $lm = @filemtime($cf);
+    $is_http = substr(DOKU_MESSAGEURL, 0, 5) != 'https';
 
     // check if new messages needs to be fetched
     if($lm < time()-(60*60*24) || $lm < @filemtime(DOKU_INC.DOKU_SCRIPT)){
         @touch($cf);
-        dbglog("checkUpdateMessages(): downloading messages to ".$cf);
-        $http = new DokuHTTPClient();
-        $http->timeout = 12;
-        $resp = $http->get(DOKU_MESSAGEURL.$updateVersion);
-        if(is_string($resp) && ($resp == "" || substr(trim($resp), -1) == '%')) {
-            // basic sanity check that this is either an empty string response (ie "no messages")
-            // or it looks like one of our messages, not WiFi login or other interposed response
-            io_saveFile($cf,$resp);
-        } else {
-            dbglog("checkUpdateMessages(): unexpected HTTP response received");
+        if(_checkUpdateMessages_remote($cf, DOKU_MESSAGEURL)){
+            // fallback to HTTP if in HTTPS
+            if(!$is_http){
+                _checkUpdateMessages_remote($cf, substr_replace(DOKU_MESSAGEURL, 'http', 0, 5));
```

Comment:
I think this fallback is not necessary. The initial check if HTTPS is supported should be enough. When ssl is available then the request should always go over HTTPS. (sorry, I might not have been clear in my initial comment).

without the fallback the whole thing gets simpler again and we should be able to keep it within the one function.

Reply:
When working on this, I referred WordPress's practice (HTTPS first, then fallback to HTTP): https://github.com/WordPress/WordPress/blob/bbb8d48086b7d10908f4fda673585ee122f2851d/wp-includes/update.php#L107-L134

If a simpler flow is preferred I will do that.